code/mob/living/critter/ai/flock/flocktasks.dm

```diff
@@ -94,6 +94,11 @@ stare
 	src.subtasks = list() //get rid of the move and replace it with flockmove
 	add_task(holder.get_instance(/datum/aiTask/succeedable/move/flock, list(holder)))
 
+/datum/aiTask/sequence/goalbased/flock/switched_to()
+	. = ..()
+	var/mob/living/critter/flock/drone/D = holder.owner
+	if(istype(D))
```

Comment:
This PR contains three different ways of casting `holder.owner` to `/mob/living/critter/flock/drone`, please gods pick one